components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/util/ActionValidatorTest.java

```diff
@@ -0,0 +1,202 @@
+                {"Username", null},
+                {"Password", ""},
+                {"Access token", null},
+                {"API key header name", null},
+                {"API key value", null}
+        };
+    }
+
+    @Test(expectedExceptions = ActionMgtClientException.class, dataProvider = "isBlankDataProvider")
+    public void testIsBlank(String fieldName, String fieldValue) throws ActionMgtClientException {
+
+        actionValidator.isBlank(fieldName, fieldValue);
+    }
+
+    @DataProvider
+    public Object[][] isNotBlankDataProvider() {
+
+        return new String[][]{
+                {"Action name", "test-action"},
+                {"Endpoint authentication URI", "https://testapi.com"},
+                {"Endpoint authentication type", "BASIC"},
+                {"Username", "testuser"},
+                {"Password", "test@123"}
+        };
+    }
+
+    @Test(dataProvider = "isNotBlankDataProvider")
+    public void testIsNotBlank(String fieldName, String fieldValue) {
+
+        try {
+            actionValidator.isBlank(fieldName, fieldValue);
+        } catch (ActionMgtClientException e) {
+            Assert.fail("Exception should not be thrown for the field: " + fieldName + " and value: " + fieldValue);
+        }
+    }
+
+    @DataProvider
+    public Object[][] invalidActionNameDataProvider() {
+
+        return new String[][]{
+                {"@test-action"},
+                {"test#action"},
+                {"test(header"},
+                {"test*header"},
+        };
+    }
+
+    @Test(expectedExceptions = ActionMgtClientException.class, dataProvider = "invalidActionNameDataProvider")
+    public void testIsInvalidActionName(String actionName) throws ActionMgtClientException {
+
+        actionValidator.isValidActionName(actionName);
+    }
+
+    @DataProvider
+    public Object[][] validActionNameDataProvider() {
+
+        return new String[][]{
+                {"test-action"},
+                {"testaction1"},
+                {"test_header"},
+                {"testHeader"},
+        };
+    }
+
+    @Test(dataProvider = "validActionNameDataProvider")
+    public void testIsValidActionName(String actionName) {
+
+        try {
+            actionValidator.isValidActionName(actionName);
+        } catch (ActionMgtClientException e) {
+            Assert.fail("Exception should not be thrown for the action name: " + actionName);
+        }
+    }
+
+    @DataProvider
+    public Object[][] invalidEndpointUriDataProvider() {
+
+        return new String[][]{
+                {"http://example.com/path?query=param"},
+                {"https://example .com "},
+                {"ftps://fileserver.com/resource"},
+                {"https://-example.com "},
+        };
+    }
+
+    @Test(expectedExceptions = ActionMgtClientException.class, dataProvider = "invalidEndpointUriDataProvider")
+    public void testIsInvalidEndpointUri(String endpointUri) throws ActionMgtClientException {
```

Comment:
Lets validate the exception message as well in here

Reply:
Updated with 487a8d524e3dd6f8618595b7b9d9266bd19f8c59